python/ros2asn1/ros2asn1.py

```diff
@@ -39,17 +39,35 @@ def process_all_messages(out_dir):
     if not os.path.exists(out_dir):
         os.makedirs(out_dir)
 
-    # Find packages with messages
-    packages = sorted([pkg for pkg, _ in rosmsg.iterate_packages(rospack, rosmsg.MODE_MSG)])
-
-    for pkg in packages:
-        print('Creating ASN.1 types for {}'.format(pkg))
-        
-        pkg_obj = RosAsn1Generator(rospack, pkg)
+    # Find packages with messages and services
+    existing_packages = [pkg for pkg, _ in rosmsg.iterate_packages(rospack, rosmsg.MODE_MSG)]
+    for pkg, _ in rosmsg.iterate_packages(rospack, rosmsg.MODE_SRV):
+        if pkg not in existing_packages:
+            existing_packages.append(pkg)
+    existing_packages = sorted(existing_packages)
 
-        # ASN.1 types "pkg.asn"
-        asn_txt = asn_template.render(pkg=pkg_obj)
+    #check that packages given in input exists
+    if packages_to_process:
+        for pkg in packages_to_process:
+            if pkg not in existing_packages:
+                print("process_all_messages : input package {} does not exist".format(pkg))
+                sys.exit(-1)
+                return false
```

Comment:
`sys.exit(-1)` exits the program entirely. I don't think `return false` is called at all.